analytical_engine/core/utils/trivial_tensor.h

```diff
@@ -69,5 +70,40 @@ struct trivial_tensor_t {
   std::vector<size_t> shape_;
   T* data_;
 };
+
+template <>
+struct trivial_tensor_t<std::string> {
+ public:
+  trivial_tensor_t() : size_(0) {}
+
+  ~trivial_tensor_t() = default;
+
+  std::vector<std::string>& data() { return data_; }
+
+  const std::vector<std::string>& data() const { return data_; }
+
+  void fill(const std::string& value) { data_.push_back(value); }
+
+  std::vector<size_t> shape() const { return shape_; }
+
+  size_t size() const { return size_; }
+
+  void resize(std::vector<size_t> const& shape) {
+    size_t flat_size = shape.empty() ? 0 : 1;
+    for (auto dim_size : shape) {
+      flat_size *= dim_size;
+    }
+    this->shape_ = shape;
+    if (flat_size != size_) {
+      data_.resize(flat_size);
+      size_ = flat_size;
+    }
+  }
+
+ private:
+  size_t size_;
+  std::vector<size_t> shape_;
+  std::vector<std::string> data_;
```

Comment:
If you only need to "Append" values, I think `arrow::StringArray` would be more suitable and more efficient.

Reply:
Updated